onyx-database/src/com/onyx/descriptor/EntityDescriptor.java

```diff
@@ -283,7 +283,25 @@ protected void validateRelationships() throws EntityClassNotFoundException, Inva
 
             if(descriptor.getType() != descriptor.getInverseClass() && (descriptor.getRelationshipType() == RelationshipType.MANY_TO_ONE || descriptor.getRelationshipType() == RelationshipType.ONE_TO_ONE))
             {
-                throw new InvalidRelationshipTypeException(InvalidRelationshipTypeException.INVERSE_RELATIONSHIP_MISMATCH);
+                throw new InvalidRelationshipTypeException(InvalidRelationshipTypeException.INVERSE_RELATIONSHIP_MISMATCH + ": " + descriptor.getParentClass().getCanonicalName());
+            }
+
+            // Verify a to many relationship is indeed a list
+            if(descriptor.getRelationshipType() == RelationshipType.ONE_TO_MANY || descriptor.getRelationshipType() == RelationshipType.MANY_TO_MANY)
+            {
+                if(descriptor.getType() != List.class)
```

Comment:
It would, I agree but, it was developed without the foresight and we are pretty dependent on List.  So for right now, I have left it as a List